packages/roosterjs-editor-core/lib/editor/Editor.ts

```diff
@@ -170,7 +174,52 @@ export default class Editor {
      * @returns true if node is inserted. Otherwise false
      */
     public insertNode(node: Node, option?: InsertOption): boolean {
-        return node ? this.core.api.insertNode(this.core, node, option) : false;
+        let darkModeNormalize = null;
+        if (this.isDarkMode()) {
+            darkModeNormalize = this.convertContentToDarkMode(node);
+        }
+
+        const result = node ? this.core.api.insertNode(this.core, node, option) : false;
+
+        if (result && darkModeNormalize) {
+            darkModeNormalize();
+        }
+        return result;
+    }
+
+    private isElement(node: Node | HTMLElement): node is HTMLElement {
+        return (<HTMLElement>node).getElementsByTagName !== undefined && (<HTMLElement>node).style !== undefined;
+    }
+
+    private isParentNode(node: Node | ParentNode): node is ParentNode {
+        return (<ParentNode>node).querySelectorAll !== undefined;
+    }
+
+    private convertContentToDarkMode(node: Node): () => void {
+        let childElements: HTMLElement[];
+        // For dark mode, we must ammass a list of all the decendents of a node.
+        // To do this, we have to take mulitple tactics, as a Node could be literally anything.
+        // If a node is an element, getElementsByTagName will return a list of HTML elements.
+        // If it's not an element, it could be a DocumentFragment. querySelectorAll has broad support and achieves the same result.

```

Comment:
This whole paragraph is written in a v. flowery way.

In previous jobs I've seen rules (and even linters) to avoid use of overly "poetic" comments / turns of phrase that are overcomplicated because they make maintaining code harder for an international team where English is not everyone's first language.

suggest something like
> Get a collection of the descendants of this node.
> If the node is not an Element or ParentNode, it has no descendants

Reply:
Also, 
https://developer.mozilla.org/en-US/docs/Web/API/ParentNode
> The ParentNode mixin contains methods and properties that are common to all types of Node objects that can have children. It's implemented by Element, Document, and DocumentFragment objects.

Do you even need the Element check?

Reply:
Yes. See my other long comment about why it's being done this way. I'll simplify the comment here and correct the logic slightly.

Reply:
Cool.

We can still simplify this to a single `instanceof ParentNode` check, can't we? since it's the unifying parent type of Document / DocumentFragment / Node.